🐛 Bug: initialize script points to wrong file

### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/create-typescript-app/issues?q=is%3Aissue) and found none that matched my issue.

### Expected

`pnpm initialize` should, you know, _work_.

### Actual

```plaintext
josh ~/repos/create-typescript-app $ pnpm initialize    

> create-typescript-app@1.52.4 initialize /Users/josh/repos/create-typescript-app
> tsx ./src/bin/index.js --mode initialize


node:internal/process/esm_loader:34
      internalBinding('errors').triggerUncaughtException(
                                ^
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/Users/josh/repos/create-typescript-app/src/bin/index.js' imported from /Users/josh/repos/create-typescript-app/
    at finalizeResolution (node:internal/modules/esm/resolve:263:11)
    at moduleResolve (node:internal/modules/esm/resolve:908:10)
    at defaultResolve (node:internal/modules/esm/resolve:1121:11)
    at nextResolve (node:internal/modules/esm/hooks:750:28)
    at d (file:///Users/josh/repos/create-typescript-app/node_modules/.pnpm/tsx@4.6.2/node_modules/tsx/dist/esm/index.mjs:5:34)
    at O (file:///Users/josh/repos/create-typescript-app/node_modules/.pnpm/tsx@4.6.2/node_modules/tsx/dist/esm/index.mjs:5:1187)
    at nextResolve (node:internal/modules/esm/hooks:750:28)
    at Hooks.resolve (node:internal/modules/esm/hooks:238:30)
    at handleMessage (node:internal/modules/esm/worker:199:24)
    at Immediate.checkForMessages (node:internal/modules/esm/worker:141:28) {
  code: 'ERR_MODULE_NOT_FOUND',
  url: 'file:///Users/josh/repos/create-typescript-app/src/bin/index.js'
}

Node.js v21.5.0
 ELIFECYCLE  Command failed with exit code 1.
josh ~/repos/create-typescript-app $ 
```

### Additional Info

Reported by @johnnyreilly in https://github.com/JoshuaKGoldberg/create-typescript-app/pull/1153#discussion_r1438932869.

--- a/package.json
+++ b/package.json
@@ -24,7 +24,7 @@
 	"scripts": {
 		"build": "tsup",
 		"format": "prettier \"**/*\" --ignore-unknown",
-		"initialize": "tsx ./src/bin/index.js --mode initialize",
+		"initialize": "tsx ./bin/index.js --mode initialize",
 		"lint": "eslint . .*js --max-warnings 0",
 		"lint:knip": "knip",
 		"lint:md": "markdownlint \"**/*.md\" \".github/**/*.md\" --rules sentences-per-line",

